backend/src/core/services/tesseract.service.ts

```diff
@@ -0,0 +1,19 @@
+// import config from '@core/config'
+
+const VAULT_DOMAIN = 'storage.vault.gov.sg'
+
+/**
+ * Validate hash for a given campaignId, recipient and version
+ */
+const validateVaultUrl = (url: string): void => {
```

Comment:
Would it be a cleaner to make this an Express middleware that responds with status 400 if the URL is invalid, and assigns the normalized URL to `res.locals.url` otherwise? I find this somewhat analogous to `getCredentialsFromLabel`: https://github.com/opengovsg/postmangovsg/blob/develop/backend/src/sms/middlewares/sms.middleware.ts#L129

Reply:
I agree that would be a neater implementation, I have refactored the `validateUrl` function in tesseract service to be a middleware within `campaign.middleware`. The